sa/sa.go

```diff
@@ -2143,3 +2148,67 @@ func (ssa *SQLStorageAuthority) IncidentsForSerial(ctx context.Context, req *sap
 	}
 	return incidentsForSerial, nil
 }
+
+// SerialsForIncident queries the provided incident table and returns the
+// resulting rows as a stream of `*sapb.IncidentSerial`s. The caller receives
+// this stream of serials by iterating over the `Recv()` method of the returned
+// streaming client. An `io.EOF` error signals that there are no more serials to
+// send. If the incident table in question contains zero rows, only an `io.EOF`
+// error is returned.
+func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentRequest, stream sapb.StorageAuthority_SerialsForIncidentServer) error {
+	if req.IncidentTable == "" {
+		return errIncompleteRequest
+	}
+
+	// Check that `req.IncidentTable` is a valid incident table name.
+	if !validIncidentTableRegexp.MatchString(req.IncidentTable) {
+		return berrors.InternalServerError("requested table name is malformed")
+	}
+
+	modelColumns := incidentSerialModel{}.columns()
+	query := fmt.Sprintf("SELECT %s FROM %s", strings.Join(modelColumns, ", "), req.IncidentTable)
+	rows, err := ssa.dbMap.Db.QueryContext(stream.Context(), query)
```

Comment:
GitHub won't let me reply to the previous comment thread, but you said:

> We're using *sql.DB.QueryContext instead of *db.WrappedMap.Query so we can pass the context.

db.WrappedMap has a `.WithContext` that allows us to pass a context. If that's the only barrier to using gorp for this, we should use that and switch to gorp.

Reply:
Yeah, I was definitely wrong there. Unfortunately, using gorp (e.g. `dbMap.WithContext().Query()`) still won't buy us any of nice field binding to a 'holder' that we get with methods like `dbMap.Select()`. That being said, we can absolutely use `dbMap.WithContext().Query()` here as a direct replacement for `dbMap.Db.QueryContext`, and I've already gone and done so.